### PR TITLE
Fix for issue 2762: Change CSV export to separate all names using semicolon

### PR DESCRIPTION
**Pull request by 125m125**
See #2762
Added a new LayoutFormatter that replaces all occurrences of " and " in the fieldText with "; ".
This formatter is now used for the author and editor fields when exporting to a csv file.

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/test/java/org/jabref/logic/exporter/CsvExportFormatTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        File tmpFile = testFolder.newFile();
+        String filename = tmpFile.getCanonicalPath();
+
+        exportFormat.performExport(databaseContext, filename, charset, entries);
+
+        List<String> lines = Files.readAllLines(tmpFile.toPath());
+        assertEquals(4, lines.size());
+        System.out.println(lines.get(3));
+        assertTrue(lines.get(1).matches("^.*,\"Someone, Van Something\",.*$"));
+        assertTrue(lines.get(2).matches("^.*,\"von Neumann, John; Smith, John; Black Brown, Peter\",.*$"));
```
Comment (stefan-kolb) [inline src/test/java/org/jabref/logic/exporter/CsvExportFormatTest.java]: Maybe you can separate this into three test. One for a single author, one for two and the last for N multiples.

File: src/test/java/org/jabref/logic/layout/format/AuthorAndToSemicolonReplacerTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        LayoutFormatter a = new AuthorAndToSemicolonReplacer();
+
+        // Empty case
+        Assert.assertEquals("", a.format(""));
+
+        // Single Names don't change
+        Assert.assertEquals("Someone, Van Something", a.format("Someone, Van Something"));
+
+        // Two names just one semicolon
+        Assert.assertEquals("John Smith; Black Brown, Peter", a
```
Comment (stefan-kolb) [inline src/test/java/org/jabref/logic/layout/format/AuthorAndToSemicolonReplacerTest.java]: Please separate to multiple tests as suggested above. One for each test case.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - Continued to redesign the user interface: this time the editor got a fresh coat of paint:
   - The buttons were changed to icons.
   - Removed the hidden feature that a double click in the editor inserted the current date.
+- All authors and editors are separated using semicolons when exporting to csv.
```
Comment (tobiasdiez) [inline CHANGELOG.md]: Please add a reference to the fixed issue (see the changelog entries below).

File: src/test/java/org/jabref/logic/exporter/CsvExportFormatTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    public void testAuthorsAreSeparatedBySemicolon() throws Exception {
+        File tmpFile = testFolder.newFile();
+        String filename = tmpFile.getCanonicalPath();
+
+        exportFormat.performExport(databaseContext, filename, charset, entries);
+
+        List<String> lines = Files.readAllLines(tmpFile.toPath());
+        assertEquals(4, lines.size());
+        System.out.println(lines.get(3));
+        assertTrue(lines.get(1).matches("^.*,\"Someone, Van Something\",.*$"));
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/exporter/CsvExportFormatTest.java]: It is probably better to replace the assertTrue with something like `assertEquals("mykey1; title1; Someone, Van Something", lines.get(1))`

File: CHANGELOG.md (hunk 3 above)
Comment (125m125) [inline CHANGELOG.md]: Should I move this entry to the section Fixed or keep it under Changes?

Comment (Siedlerchr) [inline CHANGELOG.md]: You can leave it there for the moment. Before a new version release we usually check and reorder the changelog

**Comment by tobiasdiez** [comment]
Thanks for the quick follow-up and I really like the solution with paramaterized tests (and the fact that you wrote tests in the first place!).

**Pull request merged**